src/shell.js

```diff
@@ -163,6 +176,10 @@ else if (ENVIRONMENT_IS_SHELL) {
 }
 else if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
   Module['read'] = function shell_read(url) {
+    var data = tryParseAsDataURI(url);
```

Comment:
(Same thing as in the other instance; for normal non-base64 URLs `data` will be undefined and it'll skip over the `intArrayToString` to the usual XHR steps.)